src/com/jwetherell/algorithms/mathematics/Division.java

```diff
@@ -25,6 +34,10 @@ public static final long divisionUsingRecursion(int a, int b) {
         int absA = Math.abs(a);
         int absB = Math.abs(b);
 
```

Comment:
The method will throw an exception before the result "long" is calculated avoiding unexpected results